### PR TITLE
[wasm] Fetch interpreter opcode info from the runtime instead of generating a table

### PR DESCRIPTION
**Pull request by kg**
Right now genmintops produces an opcode info table in mintops.ts that weighs in at around 64k characters. While it probably compresses OK, we don't really need it since the same information is stored in the interpreter's tables. This PR replaces it with a cache populated on demand by getting the info from the interp.

### CODE REVIEW

**Comment by BrzVlad** [comment]
Is the transition between JS and C so expensive that we can't have simple getters ?

**Comment by kg** [comment]
> Is the transition between JS and C so expensive that we can't have simple getters ?

Yeah, we'd be calling into C and unpacking a buffer once per opcode, which is going to be tens or hundreds of thousands of times

**Comment by BrzVlad** [comment]
What do you mean by unpacking a buffer ? What exactly is so expensive about calling a method like `int get_val (int index) { return table [index]; }` from JS ? To me, the simple overhead of a call doesn't sound that bad to make us duplicate data in JS

**Comment by kg** [comment]
> What do you mean by unpacking a buffer ? What exactly is so expensive about calling a method like `int get_val (int index) { return table [index]; }` from JS ? To me, the simple overhead of a call doesn't sound that bad to make us duplicate data in JS

I'll do some experiments and see if the overhead is as bad as I'm used to. I know in some cases the runtime(s) can inline C calls into JS, but sometimes there is language transition overhead. Since it's just accepting and returning an int and the function is small it should inline...

File: src/mono/mono/mini/interp/interp.c [hunk 1]
```diff
@@ -8945,4 +8945,21 @@ mono_jiterp_get_simd_opcode (int arity, int index)
...
 
+EMSCRIPTEN_KEEPALIVE void
+mono_jiterp_get_opcode_info (int opcode, void *_result)
+{
+	struct {
+		const char *name;
+		int length_u16, num_dregs, num_sregs, opargtype;
+	} *result = _result;
+	g_assert (result);
+	g_assert (opcode >= 0);
```
Comment (kotlarmilos) [inline src/mono/mono/mini/interp/interp.c:8956]: Nit: Could be easier to read if `g_assert (opcode >= 0 && opcode <= MINT_LASTOP);`.

File: src/mono/wasm/runtime/jiterpreter-trace-generator.ts [hunk 2]
```diff
@@ -1344,18 +1345,18 @@ export function generateWasmBody (
...
                         stmtText += ", ";
                     stmtText += getU16(firstSreg + (r * 2));
                 }
 
                 // print dregs
-                if (info[2] > 0)
+                if (info.dregs > 0)
                     stmtText += " -> ";
-                for (let r = 0; r < info[2]; r++) {
+                for (let r = 0; r < info.sregs; r++) {
```
Comment (kotlarmilos) [inline src/mono/wasm/runtime/jiterpreter-trace-generator.ts:1359]: It was iterating over `info[2]` which should be `info.dregs`.

**Review by vargaz — APPROVED** [review]

**Pull request merged**